lib/roby/custom_require.rb

```diff
@@ -1,27 +1,52 @@
 # frozen_string_literal: true
 
+require "pathname"
+
 # This is needed because of Ruby bug #9244
-module Kernel
-    alias syskit_original_require require
-    def require(path)
-        # Only deal with paths starting with models/, the other ones are
-        # unambiguous
-        if path =~ /^models\//
-            Roby.app.search_path.each do |app_dir|
-                full_path = File.join(app_dir, path)
-                if File.file?(full_path) || File.file?("#{full_path}.rb")
-                    return syskit_original_require(full_path)
-                end
+module Syskit
+    module KernelExtension #:nodoc:
+        def self.resolve(path)
+            match = /^([\w|-]+)\//.match(path)
+            return path if Pathname.new(path).absolute? || match.nil?
```

Comment:
- `Pathname.new(path).absolute?` is redundant w.r.t. the regexp
- you don't explicitly test for nil in ruby. Just `|| match`